tests/ignite/metrics/test_ndcg.py

```diff
@@ -0,0 +1,187 @@
+    ndcg.update([y_pred, y_true])
+    ndcg.reset()
+
+    with pytest.raises(NotComputableError, match=r"NGCD must have at least one example before it can be computed."):
+        ndcg.compute()
+
+
+@pytest.mark.parametrize(
+    "y_pred, y_true",
+    [
+        (torch.tensor([[0.1, 0.2, 0.3, 0.4, 0.5]]), torch.tensor([[1.0, 2.0, 3.0, 4.0, 5.0]])),
+        (torch.tensor([[3.7, 4.8, 3.9, 4.3, 4.9]]), torch.tensor([[2.9, 5.6, 3.8, 7.9, 6.2]])),
+        (
+            torch.tensor([[0.1, 0.2, 0.3, 0.4, 0.5], [3.7, 4.8, 3.9, 4.3, 4.9], [3.7, 4.8, 3.9, 4.3, 4.9]]),
+            torch.tensor([[1.0, 2.0, 3.0, 4.0, 5.0], [1.2, 4.5, 8.9, 5.6, 7.2], [2.9, 5.6, 3.8, 7.9, 6.2]]),
+        ),
+        (torch.tensor([[3.7, 3.7, 3.7, 3.7, 3.7]]), torch.tensor([[1.0, 2.0, 3.0, 4.0, 5.0]])),
+        (
+            torch.tensor([[3.7, 3.7, 3.7, 3.7, 3.7], [3.7, 3.7, 3.7, 3.7, 3.9]]),
+            torch.tensor([[1.0, 2.0, 3.0, 4.0, 5.0], [1.0, 2.0, 3.0, 4.0, 5.0]]),
+        ),
+    ],
+)
+@pytest.mark.parametrize("k", [None, 2, 3])
+def test_exponential(y_pred, y_true, k):
+
+    device = "cpu"
+
+    ndcg = NDCG(k=k, device=device, exponential=True)
+    ndcg.update([y_pred, y_true])
+    result_ignite = ndcg.compute()
+    result_sklearn = ndcg_score(2 ** y_true.numpy() - 1, y_pred.numpy(), k=k)
+
+    np.testing.assert_allclose(np.array(result_ignite), result_sklearn, rtol=2e-7)
+
+
+@pytest.mark.parametrize(
+    "y_pred, y_true",
+    [
+        (torch.tensor([[0.1, 0.2, 0.3, 0.4, 0.5]]), torch.tensor([[1.0, 2.0, 3.0, 4.0, 5.0]])),
+        (torch.tensor([[3.7, 4.8, 3.9, 4.3, 4.9]]), torch.tensor([[2.9, 5.6, 3.8, 7.9, 6.2]])),
+        (
+            torch.tensor([[0.1, 0.2, 0.3, 0.4, 0.5], [3.7, 4.8, 3.9, 4.3, 4.9], [3.7, 4.8, 3.9, 4.3, 4.9]]),
+            torch.tensor([[1.0, 2.0, 3.0, 4.0, 5.0], [1.2, 4.5, 8.9, 5.6, 7.2], [2.9, 5.6, 3.8, 7.9, 6.2]]),
+        ),
+    ],
+)
+@pytest.mark.parametrize("k", [None, 2, 3])
+def test_output_cpu_ignore_ties(y_pred, y_true, k):
+
+    device = "cpu"
+
+    ndcg = NDCG(k=k, device=device, ignore_ties=True)
+    ndcg.update([y_pred, y_true])
+    result_ignite = ndcg.compute()
+    result_sklearn = ndcg_score(y_true.numpy(), y_pred.numpy(), k=k)
+
+    np.testing.assert_allclose(np.array(result_ignite), result_sklearn, rtol=2e-7)
+
+
+@pytest.mark.skipif(torch.cuda.device_count() < 1, reason="Skip if no GPU")
+@pytest.mark.parametrize(
+    "y_pred, y_true",
+    [
+        (torch.tensor([[0.1, 0.2, 0.3, 0.4, 0.5]]), torch.tensor([[1.0, 2.0, 3.0, 4.0, 5.0]])),
+        (torch.tensor([[3.7, 4.8, 3.9, 4.3, 4.9]]), torch.tensor([[2.9, 5.6, 3.8, 7.9, 6.2]])),
+        (
+            torch.tensor([[0.1, 0.2, 0.3, 0.4, 0.5], [3.7, 4.8, 3.9, 4.3, 4.9], [3.7, 4.8, 3.9, 4.3, 4.9]]),
+            torch.tensor([[1.0, 2.0, 3.0, 4.0, 5.0], [1.2, 4.5, 8.9, 5.6, 7.2], [2.9, 5.6, 3.8, 7.9, 6.2]]),
+        ),
+    ],
+)
+@pytest.mark.parametrize("k", [None, 2, 3])
+def test_output_gpu_ignore_ties(y_pred, y_true, k):
```

Comment:
Maybe `ignore_ties` can be a parameter like `k`

Reply:
Actually we dont want to test the examples with ties with `ignore_ties = True`. Thats why